New fields in tls.Config in golang 1.7

When using go 1.7, `pkg/util/net/http_test.go` is failing with the following error:

```
--- FAIL: TestCloneTLSConfig (0.00s)
        http_test.go:72: New fields seen in http.Transport: [DynamicRecordSizingDisabled Renegotiation]
                Add to CopyClientTLSConfig if client-relevant, then add to expected list in TestCopyClientTLSConfig
FAIL
FAIL    k8s.io/kubernetes/pkg/util/net  0.007s
```

I'm also trouble in the issue when I execute `make test` command on release-1.4 branch of Kubernetes. Have someone fixed it? Fails (not all):
```
--- FAIL: TestCloneTLSConfig (0.00s)
	http_test.go:72: New fields seen in http.Transport: [DynamicRecordSizingDisabled Renegotiation]
		Add to CopyClientTLSConfig if client-relevant, then add to expected list in TestCopyClientTLSConfig
FAIL
FAIL	k8s.io/kubernetes/pkg/util/net	0.056s
```
```
root@heatonli-test1:/home/zhangjian/src/k8s.io/kubernetes# go version
go version go1.7.1 linux/amd64
```

--- a/pkg/util/net/http_test.go
+++ b/pkg/util/net/http_test.go
@@ -23,6 +23,8 @@ import (
 	"net/url"
 	"os"
 	"reflect"
+	"runtime"
+	"strings"
 	"testing"
 
 	"k8s.io/kubernetes/pkg/util/sets"
@@ -52,14 +54,17 @@ func TestCloneTLSConfig(t *testing.T) {
 		// These fields are not copied
 		"SessionTicketsDisabled",
 		"SessionTicketKey",
-		"DynamicRecordSizingDisabled",
-		"Renegotiation",
 
 		// These fields are unexported
 		"serverInitOnce",
 		"mutex",
 		"sessionTicketKeys",
 	)
+
+	// See #33936.
+	if strings.HasPrefix(runtime.Version(), "go1.7") {
+		expected.Insert("DynamicRecordSizingDisabled", "Renegotiation")
+	}
 
 	fields := sets.NewString()
 	structType := reflect.TypeOf(tls.Config{})

